api/src/client/ApiTaxClearanceCertificateClient.ts

```diff
@@ -0,0 +1,114 @@
+  const postTaxClearanceCertificate = async (
+    userData: UserData
+  ): Promise<TaxClearanceCertificateResponse> => {
+    const logId = logWriter.GetId();
+    logWriter.LogInfo(`Tax Clearance Certificate Client - Id:${logId}`);
+
+    const currTaxClearanceData = userData.businesses[userData.currentBusinessId].taxClearanceCertificateData;
+
+    const postBody = {
+      repId: userData.user.id,
+      repName: userData.user.name,
+      taxpayerId: currTaxClearanceData.taxId,
+      taxpayerPin: currTaxClearanceData.taxPin,
+      taxpayerName: currTaxClearanceData.businessName,
+      addressLine1: currTaxClearanceData.addressLine1,
+      addressLine2: currTaxClearanceData.addressLine2,
+      city: currTaxClearanceData.addressCity,
+      state: currTaxClearanceData.addressState?.shortCode,
+      zip: currTaxClearanceData.addressZipCode,
+      agencyName: LookupTaxClearanceCertificateAgenciesById(currTaxClearanceData.requestingAgencyId).name,
+    };
+
+    logWriter.LogInfo(
+      `Tax Clearance Certificate Client - Id:${logId} - Request Sent to ${
+        config.orgUrl
+      }/TYTR_ACE_App/ProcessCertificate/businessClearance data: ${JSON.stringify(postBody)}`
+    );
+
+    return axios
+      .post(`${config.orgUrl}/TYTR_ACE_App/ProcessCertificate/businessClearance`, postBody, {
+        auth: {
+          username: config.userName,
+          password: config.password,
+        },
+      })
+      .then((response: AxiosResponse) => {
+        logWriter.LogInfo(
+          `Tax Clearance Certificate Client - Id:${logId} - Response received: ${JSON.stringify({
+            ...response.data,
+            certificate: "Succussful Response - PDF data omitted",
+          })}`
+        );
+        return { certificatePdfArray: response.data.certificate };
+      })
+      .catch((error: AxiosError) => {
+        logWriter.LogError(`Tax Clearance Certificate Client - Id:${logId} - Error`, error);
+        const { response } = error;
+        if (response?.status === StatusCodes.BAD_REQUEST) {
+          const errorMessage = response.data as string;
+          if (errorMessage === INELIGIBLE_TAX_CLEARANCE_FORM) {
+            return {
+              error: {
+                type: "INELIGIBLE_TAX_CLEARANCE_FORM",
+                message: errorMessage,
+              },
+            };
+          }
+          if (errorMessage === MISSING_FIELD || errorMessage === TAX_ID_MISSING_FIELD) {
+            return {
+              error: {
+                type: "MISSING_FIELD",
+                message: errorMessage,
+              },
+            };
+          }
+          if (errorMessage === FAILED_TAX_ID_AND_PIN_VALIDATION) {
+            return {
+              error: {
+                type: "FAILED_TAX_ID_AND_PIN_VALIDATION",
+                message: errorMessage,
+              },
+            };
+          }
+          if (errorMessage === SYSTEM_ERROR) {
+            return {
+              error: {
```

Comment:
If a Natural Program Error is received, we treat it as a SYSTEM_ERROR and return a structured error object. For any other unexpected internal errors, we allow the error to propagate by throwing it.